adoc/extensions/sycl_khr_includes.adoc

```diff
@@ -0,0 +1,610 @@
+|Contains definition of [code]#interop_handle# class
+
+|[code]#<sycl/khr/includes/backend>#
+|Contains definition of [code]#backend_traits# and [code]#make_*# functions
+
+|[code]#<sycl/khr/includes/bit>#, [code]#<sycl/khr/includes/span>#
+|Headers in this set match by name with those provided by the next C++
+specification. They contain pre-adopted features from that standard.
+
+|[code]#<sycl/khr/includes/byte>#
+|Contains definition of [code]#byte# alias
+|====
+
+Classes, constants, types and functions provided by the headers listed above
+exist within the [code]#::sycl# namespace as with the standard
+[code]#<sycl/sycl.hpp># header.
+
+Note: there is no strict requirement for the content of those headers, i.e. they
+may contain other functions and classes than documented in this extension. In
+fact, some classes, functions or enumerations are documented to be available
+through multiple headers. Nevertheless, implementations are encouraged to keep
+headers contents as small as reasonably possible to provide maximum benefit for
+users.
+
+[[sec:khr-includes-version]]
+=== [code]#<sycl/khr/includes/version># header
+
+This header contains definitions of:
+- [code]#SYCL_KHR_INCLUDES# macro
+- All macro defined by <<sec:preprocessor-definitions-and-macro>>
+- All macro defined by other extensions supported by an implementation
+
+[code]#<sycl/khr/includes/version># header is included by every other header
+documented by this extension, as well as [code]#<sycl/sycl.hpp>#.
+
+There is no guarantee that the aforementioned macro are defined before the
+inclusion point of [code]#<sycl/khr/includes/version>#.
+
+{note}Implementations providing a dedicated SYCL compiler can predefine those
+macro within the compiler. However, the SYCL standard allows for library-only
+implementations and therefore a portable application should not assume that
+any macro definitions are available until [code]#<sycl/khr/includes/version># is
+included.{endnote}
+
+To properly detect in a portable way if this extension is supported by an
+implementation, the following approach should be used:
+
+[source]
+----
+#if __has_include(<sycl/khr/includes/version>)
+// The extension is supported, but the SYCL_KHR_INCLUDES is not
+// guaranteed to be defined at this point for library-only implementations
+
+#include <sycl/khr/includes/version>
+// Now the SYCL_KHR_INCLUDES macro is guaranteed to be defined for all kinds of
+// implementations
+
+// Use the extension
+#include <sycl/khr/includes/vec>
+
+#else
+// Extension is not supported by an implementation
+// Fallback to the core SYCL standard
+#include <sycl/sycl.hpp>
+#endif
+----
+
+[[sec:khr-includes-backend]]
+=== [code]#<sycl/khr/includes/backend># header
+
+This header file contains everything related to SYCL backend APIs:
+
+- [code]#enum class backend#
+- [code]#backend_traits#
+- [code]#get_native#
+- [code]#make_*#
+
+Note that even though functions defined in the header operate with various
```

Comment:
Some kind of leftover from a previous version. The intent for this note is to apply to all headers, i.e. the extension does not mandate a strict split. It is a valid (even though useless) implementation option to just include existing `<sycl/sycl.hpp>` from every new header.

Therefore, the note should probably be moved into some common section and generalized.

Also highlighting this to double-check that everyone is on board with the suggested non-strict approach.

Reply:
I agree that we should go with the non-strict approach.  I don't think there is precedent anywhere for "if you include X, I promise you won't also get Y", it would limit what implementations can do, and it would be really hard to test.  I think phrasing that says "You must include `<sycl/sycl.hpp>` or `<sycl/header>` to _guarantee_ features X, Y and Z are available" makes the most sense.

Reply:
Generalized this note in ba0812d5b01c7528617baae93bddef40b6fe03df, also feedback on its wording is still welcome